os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_ota_ram.c

```diff
@@ -75,4 +75,121 @@ u32 OTA_Change(u32 OTAIdx)
 	return _TRUE;
 }
 
+#define TIZENRT_KERNEL_HEADER_LEN 0x10
+IMAGE2_RAM_TEXT_SECTION
+static u8 OTA_AddrChk(u8 idx, uint32_t pAddr, uint32_t *size, uint32_t *SBAddr)
+{
+	IMAGE_HEADER *Img2Hdr, *Img2DataHdr, *PsramHdr;
+	uint32_t ImgSize, SBheader;
+	u8 res = _TRUE;
+
+	/* Verify signature */
+	if(idx == 0) {
+		Img2Hdr = (IMAGE_HEADER *)(pAddr + TIZENRT_KERNEL_HEADER_LEN);
+	} else {
+		Img2Hdr = (IMAGE_HEADER *)pAddr;
+	}
+	if((Img2Hdr->signature[0] != 0x35393138) || (Img2Hdr->signature[1] != 0x31313738) || ((Img2Hdr->image_size) & 0x3)) {
+		res = _FALSE;
+		goto exit;
+	}
+
+	/* Get image size, which should be 4KB-aligned */
+	if(idx == 0) {
+		Img2DataHdr = (IMAGE_HEADER *)(pAddr + TIZENRT_KERNEL_HEADER_LEN + IMAGE_HEADER_LEN + Img2Hdr->image_size);
+		ImgSize = Img2Hdr->image_size + Img2DataHdr->image_size + IMAGE_HEADER_LEN * 2 + TIZENRT_KERNEL_HEADER_LEN;
+	} else {
+		Img2DataHdr = (IMAGE_HEADER *)(pAddr + IMAGE_HEADER_LEN + Img2Hdr->image_size);
+		if((Img2DataHdr->signature[0] != 0x35393138) || (Img2DataHdr->signature[1] != 0x31313738) || ((Img2DataHdr->image_size) & 0x3)) {
+			res = _FALSE;
+			goto exit;
+		}
+		ImgSize = Img2Hdr->image_size + Img2DataHdr->image_size + IMAGE_HEADER_LEN * 2;
+	}
+
+	/* KM4 image contains PSRAM code+data part */
+	if(idx == 1) {
+		PsramHdr = (IMAGE_HEADER *)(pAddr + ImgSize);
+		if((PsramHdr->signature[0] != 0x35393138) || (PsramHdr->signature[1] != 0x31313738) || ((PsramHdr->image_size) & 0x3)) {
+			res = _FALSE;
+			goto exit;
+		}
+		ImgSize += (IMAGE_HEADER_LEN + PsramHdr->image_size);
+
+		/*check if KM4 image contains secure image2 signature*/
+		SBheader = Img2Hdr->sb_header;
+		if (SBheader != 0xFFFFFFFF) {
+			ImgSize += 0x1000;
+			if(SBAddr != NULL)
+				*SBAddr = SBheader;
+		}
+	}
+
+	ImgSize = (((ImgSize -1) >> 12) + 1) << 12;  /* 4KB aligned */
+	if(size != NULL)
+		*size = ImgSize;
+
+exit:
+	return res;
+}
+
+IMAGE2_RAM_TEXT_SECTION
+uint32_t OTA_Image2SignatureCheck(uint32_t input_addr)
+{
+	int ret, index = 0;
+	uint32_t image2_addr, SBAddr, xip_AddrSrt, size;
+	unsigned char image_key[32];
+
+	rtw_printf("Image Start Addrs: 0x%x \n", input_addr);
+	ret = OTA_AddrChk(0, input_addr, &size, NULL);	/* KM0 Image2 Check */
+	if(ret == _FALSE)
+		goto SBOOT_FAIL;
+	rtw_printf("OTA_AddrChk: 0, size: 0x%x \n", size);
```

Comment:
`size` is unsigned int type.
Is there any special reason to print it with `%x` format?
If not, how about using `%u`?

Reply:
size will be change by OTA_AddrChk(0, it will be the size of image, we usually represent image size with hex.
So it is better to print as hex.